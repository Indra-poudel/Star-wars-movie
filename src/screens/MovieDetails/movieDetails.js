import React, {useEffect, useState} from 'react';
import {Text, ImageBackground, StyleSheet, View, FlatList} from 'react-native';

import ScreenWrapper from '../../components/screenWrapper';

import Images from '../../assets/image';

import {getIMDbRating} from '../../services/imdb';
import CharacterCard from '../../components/characterCard/characterCard';

const MovieDetails = props => {
  const {params} = props.route;
  const [imdbRating, setImdbRating] = useState('');
  const [isImdbFetching, setIsImdbFetching] = useState(true);

  useEffect(() => {
    const imdbId = Images[params.episode_id].id;
    getIMDbRating(imdbId)
      .then(imdbRating => {
        setImdbRating(imdbRating);
      })
      .catch(() => {
        setImdbRating(`Error while fetching Imdb rating`);
      })
      .finally(() => {
        setIsImdbFetching(false);
      });
  }, [params.episode_id]);

  return (
    <ScreenWrapper style={{paddingHorizontal: 0}}>
      <ImageBackground
        source={Images[params.episode_id].image}
        style={Style.backgroundImage}>
        <View style={Style.imdbWrapper}>
          {isImdbFetching ? (
            <Text style={{}}>Loading...</Text>
          ) : (
            <Text style={{}}>IMDb : {imdbRating}</Text>
          )}
        </View>
        <View style={Style.directedWrapper}>
          <Text style={Style.subtitle}>Directed by {params.director}</Text>
        </View>
      </ImageBackground>

      <View style={Style.titleWrapper}>
        <View>
          <Text style={Style.title}>{params.title}</Text>
          <Text style={Style.subtitle}>Released on {params.release_date}</Text>
        </View>
      </View>

      <Text style={Style.secondTitle}>List of Characters</Text>
      <FlatList
        style={Style.contentWrapper}
        showsHorizontalScrollIndicator={false}
        data={params.characters}
        keyExtractor={item => item}
        renderItem={({item, index}) => {
          return <CharacterCard key={index} url={item} />;
        }}
      />
    </ScreenWrapper>
  );
};

const Style = StyleSheet.create({
  backgroundImage: {
    position: 'relative',
    resizeMode: 'contain',
    overflow: 'hidden',
    flex: 0.5,
  },

  contentWrapper: {
    flex: 0.5,
  },

  titleWrapper: {
    padding: 16,
    backgroundColor: 'black',
    color: 'white',
    display: 'flex',
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
  },

  imdbWrapper: {
    padding: 10,
    borderRadius: 8,
    position: 'absolute',
    left: 10,
    bottom: 10,
    backgroundColor: 'yellow',
    color: 'black',
    fontSize: 14,
    fontWeight: 'bold',
  },

  title: {
    fontSize: 24,
    fontWeight: 'bold',
    color: 'white',
    fontFamily: 'ProximaNova-Regular',
  },

  subtitle: {
    marginTop: 12,
    fontSize: 14,
    color: '#C8C8C8',
    fontFamily: 'ProximaNova-Regular',
  },

  directedWrapper: {
    padding: 4,
    position: 'absolute',
    left: 10,
    top: 10,
    backgroundColor: 'rgba(52, 52, 52, 0.8)',
  },

  secondTitle: {
    padding: 12,
    color: 'white',
    backgroundColor: 'rgba(52, 52, 52, 0.8)',
  },
});

export default MovieDetails;
