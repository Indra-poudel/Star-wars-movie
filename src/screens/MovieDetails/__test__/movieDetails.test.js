import React from 'react';
import renderer from 'react-test-renderer';
import MovieDetails from '../movieDetails';

jest.mock('@react-navigation/elements', () => ({
  useHeaderHeight: () => 10,
}));

test('Renders snapshot as expected', () => {
  const tree = renderer
    .create(
      <MovieDetails
        route={{
          params: {
            characters: [],
            title: 'A New Hope',
            episode_id: 4,
            director: 'George Lucas',
            producer: 'Gary Kurtz, Rick McCallum',
            release_date: '1977-05-25',
          },
        }}
      />,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
