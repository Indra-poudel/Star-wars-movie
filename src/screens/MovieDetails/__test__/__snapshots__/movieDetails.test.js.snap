// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders snapshot as expected 1`] = `
<View
  style={
    Array [
      Object {
        "flexGrow": 1,
        "marginTop": 10,
        "paddingHorizontal": 10,
      },
      Object {
        "paddingHorizontal": 0,
      },
    ]
  }
>
  <View
    accessibilityIgnoresInvertColors={true}
    style={
      Object {
        "flex": 0.5,
        "overflow": "hidden",
        "position": "relative",
        "resizeMode": "contain",
      }
    }
  >
    <Image
      source={
        Object {
          "process": [Function],
        }
      }
      style={
        Array [
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          Object {
            "height": undefined,
            "width": undefined,
          },
          undefined,
        ]
      }
    />
    <View
      style={
        Object {
          "backgroundColor": "yellow",
          "borderRadius": 8,
          "bottom": 10,
          "color": "black",
          "fontSize": 14,
          "fontWeight": "bold",
          "left": 10,
          "padding": 10,
          "position": "absolute",
        }
      }
    >
      <Text
        style={Object {}}
      >
        Loading...
      </Text>
    </View>
    <View
      style={
        Object {
          "backgroundColor": "rgba(52, 52, 52, 0.8)",
          "left": 10,
          "padding": 4,
          "position": "absolute",
          "top": 10,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#C8C8C8",
            "fontFamily": "ProximaNova-Regular",
            "fontSize": 14,
            "marginTop": 12,
          }
        }
      >
        Directed by 
        George Lucas
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "black",
        "color": "white",
        "display": "flex",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "padding": 16,
      }
    }
  >
    <View>
      <Text
        style={
          Object {
            "color": "white",
            "fontFamily": "ProximaNova-Regular",
            "fontSize": 24,
            "fontWeight": "bold",
          }
        }
      >
        A New Hope
      </Text>
      <Text
        style={
          Object {
            "color": "#C8C8C8",
            "fontFamily": "ProximaNova-Regular",
            "fontSize": 14,
            "marginTop": 12,
          }
        }
      >
        Released on 
        1977-05-25
      </Text>
    </View>
  </View>
  <Text
    style={
      Object {
        "backgroundColor": "rgba(52, 52, 52, 0.8)",
        "color": "white",
        "padding": 12,
      }
    }
  >
    List of Characters
  </Text>
  <RCTScrollView
    data={Array []}
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    showsHorizontalScrollIndicator={false}
    stickyHeaderIndices={Array []}
    style={
      Object {
        "flex": 0.5,
      }
    }
    viewabilityConfigCallbackPairs={Array []}
  >
    <View />
  </RCTScrollView>
</View>
`;
